### PR TITLE
Update django

### PR DESCRIPTION
**Pull request by hramezani**

### CODE REVIEW

File: auditlog/models.py [hunk 1]
```diff
@@ -418,7 +418,7 @@ def bulk_related_objects(self, objs, using=DEFAULT_DB_ALIAS):
 try:
     from south.modelsinspector import add_introspection_rules
 
-    add_introspection_rules([], ["^auditlog\.models\.AuditlogHistoryField"])
+    add_introspection_rules([], [r"^auditlog\.models\.AuditlogHistoryField"])
```
Comment (Linkid) [inline auditlog/models.py:421]: Already done in PR #197.

File: setup.py [hunk 2]
```diff
@@ -32,7 +32,6 @@
         "Programming Language :: Python :: 3.10",
         "Framework :: Django",
         "Framework :: Django :: 2.2",
-        "Framework :: Django :: 3.1",
         "Framework :: Django :: 3.2",
```
Comment (Linkid) [inline setup.py:35]: You can add the django 4.0 classifier ;).

File: tox.ini [hunk 3]
```diff
@@ -1,7 +1,8 @@
 [tox]
 envlist =
-    {py37,py38,py39}-django{22,31}
+    {py37,py38,py39}-django22
     {py37,py38,py39,py10}-django32
+    {py38,py39,py10}-djangomain
```
Comment (Linkid) [inline tox.ini:5]: I think it should be better to explicit the version (`django40`) to know what it is going on without checking another repo.

File: tox.ini [hunk 4]
```diff
@@ -11,8 +12,8 @@ commands =
     coverage xml
 deps =
     django22: Django>=2.2,<2.3
-    django31: Django>=3.1,<3.2
     django32: Django>=3.2,<3.3
+    djangomain: https://github.com/django/django/archive/main.tar.gz
```
Comment (Linkid) [inline tox.ini:16]: Idem: replace `djangomain` with `django40` to be more explicit.

**Review by Linkid — CHANGES_REQUESTED** [review]
Great !

File: auditlog/models.py (hunk 1 above)
Comment (hramezani) [reply auditlog/models.py:421]: I think we can close that one.

File: setup.py (hunk 2 above)
Comment (hramezani) [reply setup.py:35]: Added!

File: tox.ini (hunk 3 above)
Comment (hramezani) [reply tox.ini]: I added Django4.0 to the test pipeline.

**Comment by hramezani** [comment]
Thanks @Linkid for the review. I've addressed your comments!

File: auditlog/models.py (hunk 1 above)
Comment (Linkid) [reply auditlog/models.py:421]: Hmm… I think it is better to merge other's PR instead of redoing it. Check #342 for a quick rebase.

Comment (hramezani) [reply auditlog/models.py:421]: you exactly did what I did here :)
The change is already part of this PR. why do we need to have another PR for it?

Comment (Linkid) [reply auditlog/models.py:421]: To give credit to the one who did the fix. But if you think it is better to include it in your PR, let's go.

**Review by Linkid — APPROVED** [review]

**Pull request merged**